### PR TITLE
Operators: AcquiaSnapshot + AcquiaSnapshotScheduled

### PR DESCRIPTION
**Pull request by nickschuch**
* Splits out binaries into `mtk` and `mtk-operator`
* Adds examples

### CODE REVIEW

File: cmd/mtk-operator/acquia/scheduled.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	sdk.Run(context.TODO())
+	return nil
+}
+
+// SnapshotScheduled declares the "operator-scheduled" subcommand.
+func SnapshotScheduled(app *kingpin.CmdClause) {
+	c := new(cmdSnapshotScheduled)
+
+	cmd := app.Command("scheduled", "Operator for running a single 'Scheduled' snapshot").Action(c.run)
+	cmd.Flag("watch", "Namespace to watch for new AcquiaSnapshot objects").Default(corev1.NamespaceAll).Envar(envar.OperatorWatch).StringVar(&c.Watch)
```
Comment (nicksantamaria) [inline cmd/mtk-operator/acquia/scheduled.go]: `--watch` is usually used in boolean contexts - can we change this to `--namespace-watch` (or something similar) ?

File: cmd/mtk-operator/acquia/scheduled.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	cmd.Flag("namespace", "Namespace to execute AcquiaSnapshot Jobs").Required().Envar(envar.OperatorNamespace).StringVar(&c.Namespace)
```
Comment (nicksantamaria) [inline cmd/mtk-operator/acquia/scheduled.go]: Maybe (or maybe not) rename based on feedback in comment above.

File: cmd/mtk-operator/acquia/scheduled.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+	cmd.Flag("secret", "Secret for loading Operator configuration").Required().Envar(envar.OperatorSecret).StringVar(&c.Secret)
+	cmd.Flag("resync", "How often to resync all the AcquiaScheduled objects").Default("30").Envar(envar.OperatorResync).IntVar(&c.Resync)
+	cmd.Flag("job-image", "Image to use when executing a Job").Default("previousnext/mtk:latest").Envar(envar.OperatorJobImage).StringVar(&c.JobImage)
+	cmd.Flag("job-cpu", "CPU allocated when executing of a Job").Default("250m").Envar(envar.OperatorJobCPU).StringVar(&c.JobCPU)
```
Comment (nicksantamaria) [inline cmd/mtk-operator/acquia/scheduled.go]: Is there validation for this somewhere? i.e. what happens if an invalid value is provided here?

Comment (nicksantamaria) [reply cmd/mtk-operator/acquia/scheduled.go]: Could probably be implemented in a future PR.

File: example/mtk.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    # Drupal 8
+    - name: users_field_data
+      fields:
+        - name: mail
+          value: "SANITIZED_MAIL"
+        - name: pass
+          value: "SANITIZED_PASSWORD"
+
+nodata:
+  regex:
```
Comment (nicksantamaria) [inline example/mtk.yml]: This is still a thing?

File: internal/operator/handler/acquia/secrets/types.go [hunk 5]
```diff
@@ -0,0 +1,13 @@
+package secrets
+
+// Secrets used for Acquia to AWS CodeBuild to Docker Hub.
+type Secrets struct {
```
Comment (nicksantamaria) [inline internal/operator/handler/acquia/secrets/types.go]: 🤤

File: example/mtk.yml (hunk 4 above)
Comment (nickschuch) [reply example/mtk.yml]: Nope!

File: cmd/mtk-operator/acquia/scheduled.go (hunk 3 above)
Comment (nickschuch) [reply cmd/mtk-operator/acquia/scheduled.go]: Deep down when we generate the Job.

https://github.com/previousnext/mysql-toolkit/blob/acquia-scheduled/internal/operator/handler/acquia/job/job.go#L171

File: example/mtk.yml [hunk 6]
```diff
@@ -16,20 +16,18 @@ sanitize:
...
-    - flood
-    - batch
-    - queue
-    - sessions
-    - semaphore
-    - search_api_task
-    - search_dataset
-    - search_index
-    - search_total
+  - cache.*
```
Comment (nicksantamaria) [inline example/mtk.yml]: If this is using globbing now, shouldn't it be `cache*` ? (no period).

**Pull request merged**